cilium/CFP-26135-default-egress-allow.md

```diff
@@ -0,0 +1,198 @@
+
+##### Links
+https://github.com/kubernetes/enhancements/issues/2091
+https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2091-admin-network-policy 
+https://github.com/cilium/cilium/issues/23380
+
+
+## Impacts / Key Questions
+
+### Key Question: Why commandLine Flag Option not CRDs?
+Two approaches are being considered to pass input for CloudProvider policies: Custom Resource Definitions (CRD) and Cilium configuration (command-line).
+
+Passing input via CRD: As cloud provider policies are cluster scoped, `CiliumClusterWideNetworkPolicy` is the right option to use. With this approach, it may require addition of new field in `CiliumClusterWideNetworkPolicy` CRD to differentiate cloud provider policy from other network policies. This in turn requires addition of new field in `api:Rule`as cilium parses all Network Policies to `api:Rule`.A potential concern with this method is that cilium uses same `api:Rule` for NetworkPolicy and CiliumNetworkPolicy CRDs as well. Since this feature is supported only with `CiliumClusterWideNetworkPolicy`, adding a new field might confuse customers when using other Network Policies. The other concern is that cluster admin is expected to have control over all `CiliumClusterwideNetworkPolicy` except for this cloud provider specific policy. This will complicate the RBAC model between cloud admin and cluster admin when managing these policies
+
+Passing input via Cilium configuration (command-line): Cloud provider policies doesn’t require all features present in other network policies. It doesn't change any existing or any new CRDs and can be kept away from regular users. Additionally, since these policies are not expected to undergo frequent updates, handling them through the Cilium configuration offers a reasonable and manageable approach.
+
+### Key Question: How cloud provider deny rules are processed with network policy rules for same endpoint?
+There is no change to default deny behavior of existing network policies even if cloud provider specific deny rules configured via cilium config.  As EgressDeny rules are processed first, deny rules from cloud provider are evaluated first and rest of rules are evaluated later as like other network policy egress deny rules.
+
+### Key Question: Why not iptable rules instead of eBPF?
+cilium adds a rule on top of the FORWARD chain to redirect all traffic to a CILIUM_FORWARD chain. So any Cloud Provider specific firewall rules placed after cilium rules doesn't get executed. Disabling the "prepend-iptables-chains" option is also not a viable solution as disabling prepend means Cilium can no longer guarantee that its policies are enforced. Also in eBPF host routing mode, Cilium bypasses IPtable rules and rules can be enforced only via eBPF. 
```

Comment:
nit: These are good context, but for context usually the goal of "Key Questions" is to ask the reviewers to consider the options and provide input on the solutions. These two seem to be more of FAQ items which are probably more suited to the implementatino section.